src/tasks/deploy-sync.js

```diff
@@ -28,10 +29,16 @@ gulp.task('deploy:sync-init', () => {
   const environment = config.environment.split(/\s*,\s*|\s+/)[0];
 
   const envObj = tkConfig[environment];
+  const valid_id = utils.validateThemeId(envObj.theme_id);
   let proxyTarget = `https://${envObj.store}`;
 
-  if (envObj.theme_id && (envObj.theme_id === parseInt(envObj.theme_id, 10))) {
-    proxyTarget += `?preview_theme_id=${envObj.theme_id}`;
+  if (!valid_id) {
+    messages.invalidThemeId(envObj.theme_id, environment);
+    process.exit();
```

Comment:
In slate-cli (https://github.com/Shopify/slate-cli/blob/master/src/index.js#L77), I tried to make sure code would only run if all checks passed; but code wouldn't stop execution... if one of the checks failed, other code could still run. Whereas if I used `process.exit()`, I might end up stopping execution accidentally early.

Maybe this check should happen earlier in the code? Like would we ever let someone run watch if they had an invalid theme id?